nepalingo-web/src/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSearch } from '@fortawesome/free-solid-svg-icons';
+import useDictionary, { DictionaryProps, DictionaryResponse } from '../../hooks/useDictionary';
+
+const SearchBar: React.FC = () => {
+  const [searchTerm, setSearchTerm] = useState<string>('');
+  const language: DictionaryProps['language'] = 'newari'; 
+
+  const { data, isLoading, error } = useDictionary({ language, word: searchTerm });
+  console.log('Fetched data:', data);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchTerm(e.target.value);
+  };
+
+  const handleSearchClick = () => {
+    // Trigger re-fetch (not necessary for useSWR as it auto-updates)
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter') {
+      handleSearchClick();
+    }
+  };
+
+  return (
+    <div className="flex flex-col items-center mt-5">
+      <div className="w-3/4 flex items-center relative">
+        <input
+          type="text"
+          value={searchTerm}
+          onChange={handleChange}
+          onKeyDown={handleKeyDown}
+          placeholder="Search here"
+          className="w-full p-2 pl-5 border border-gray-300 rounded-md text-lg shadow-sm transition-colors duration-300 focus:border-blue-500 focus:shadow-lg bg-white"
+        />
+        <span
+          className="absolute right-2 text-lg text-gray-600 cursor-pointer"
+          onClick={handleSearchClick}
+        >
+          <FontAwesomeIcon icon={faSearch} />
+        </span>
+      </div>
+
+      {error && <p className="mt-2 text-red-600">{error.message}</p>}
+
+
+
+
+
+      <ul className="list-none p-0 mt-5 w-3/4">
+  {isLoading && <p className="mt-2 text-gray-600">Loading...</p>}
+  {data && data.meanings ? (
+    data.meanings.length > 0 ? (
+      data.meanings.map((meaning, index) => (
+        <li key={index} className="bg-gray-100 p-4 mb-2 rounded-md shadow transition-all duration-300 hover:bg-gray-200">
```

Comment:
You shouldn't use the index as your key. Here's a writeup of why: https://chatgpt.com/share/ff75ba65-6c8a-4076-b1c0-5e5ad1576ff2

Reply:
Should she use `meaning.meaningNp` here then?

Reply:
I would use the original word

Reply:
I don't understand this point. What do you mean by 'translated meaning'?